File: SmoothDescentPeriods.java
// https://leetcode.com/problems/number-of-smooth-descent-periods-of-a-stock/description/?envType=daily-question&envId=2025-12-15

class SmoothDescentPeriods{
    public long getDescentPeriods(int[] prices){
        int n = prices.length;
        long res = 1;
        int prev = 1;

        for(int i = 1; i < n; i++){
            if(prices[i] == prices[i - 1] - 1){
                prev++;
            }
            else{
                prev = 1;
            }
            res = res + prev;
        }
        return res;
    }

    public static void main(String[] args) {
        SmoothDescentPeriods sdp = new SmoothDescentPeriods();
        int[] prices = {3, 2, 1, 4};
        long result = sdp.getDescentPeriods(prices);
        System.out.println("Total smooth descent periods: " + result);
    }
}

File: Python/SmoothDescentPeriods.py
# https://leetcode.com/problems/number-of-smooth-descent-periods-of-a-stock/?envType=daily-question&envId=2025-12-15

from typing import List

class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        n = len(prices)
        res = 1
        prev = 1

        for i in range(1, n):
            if prices[i] == prices[i - 1] - 1:
                prev += 1
            else:
                prev = 1
            res += prev
        return res
    
if __name__ == "__main__":
    prices = list(map(int, input().split()))
    solution = Solution()
    result = solution.getDescentPeriods(prices)
    print(result)
